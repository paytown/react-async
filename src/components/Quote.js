import React from 'react';
import PropTypes from 'prop-types';

function Quote({ quote }) {
  return (
    <div>
      <img src={quote.image} />
      <p>{quote.character}</p>
      <p>{quote.quote}</p>
    </div>
  );
}

Quote.propTypes = {
  quote: PropTypes.shape({
    image: PropTypes.string.isRequired,
    character: PropTypes.string.isRequired,
    quote: PropTypes.string.isRequired,
  })
};

export default Quote;
