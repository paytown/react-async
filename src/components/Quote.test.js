import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote';

describe('quote', () => {
  it('displays quotes', () => {
    const quote = {
      character: 'Bob',
      quote: 'no',
      image: '/img'
    };
    const wrapper = shallow(<Quote quote={quote} />);
    expect(wrapper).toMatchSnapshot();
  });
});
