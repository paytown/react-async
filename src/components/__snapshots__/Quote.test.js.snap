// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quote displays quotes 1`] = `
<div>
  <img
    src="/img"
  />
  <p>
    Bob
  </p>
  <p>
    no
  </p>
</div>
`;
